engines/optix/OptiXModel.h

```diff
@@ -114,6 +121,19 @@ class OptiXModel : public Model
     std::map<std::string, optix::Buffer> _optixTextures;
     std::map<std::string, optix::TextureSampler> _optixTextureSamplers;
 
+    // Transfer function
+    struct OptiXTransferFunction
+    {
+        optix::Buffer colors{nullptr};         // vec3fs
+        optix::Buffer opacities{nullptr};      // floats
+        optix::Buffer simulationData{nullptr}; // floats
+        bool initialized{false};
+    };
+
+    OptiXTransferFunction _tf;
```

Comment:
If this type is not used anywhere else, you can use an unnamed struct type as well:

    struct
    {
        ....
    } _transferFunction;